src/term/mod.rs

```diff
@@ -1305,6 +1305,9 @@ pub enum BinaryOp {
     /// Lazily apply a contract to an Array.
     /// This simply inserts a contract into the array attributes.
     ArrayLazyAssume(),
+
+    /// TODO
```

Comment:
Does it make sense to document this before merging?

Reply:
Good catch, that should definitely get a proper got-docmment.